packages/orchestrator-ui-components/src/components/WfoRadioDropdown/WfoRadioDropdown.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from 'react';
+
+import {
+    EuiButtonEmpty,
+    EuiFlexGroup,
+    EuiPopover,
+    EuiRadioGroup,
+} from '@elastic/eui';
+import { EuiRadioGroupOption } from '@elastic/eui/src/components/form/radio/radio_group';
+
+import { WfoRadioDropdownOption } from '@/types';
+
+export interface WfoRadioDropdownProps<T> {
+    options: WfoRadioDropdownOption<T>[];
+    selectedOption: WfoRadioDropdownOption<T>;
+    onUpdateOption: (value: WfoRadioDropdownOption<T>) => void;
+}
+
+export const WfoRadioDropdown = <T,>({
+    options,
+    onUpdateOption,
+    selectedOption,
+}: WfoRadioDropdownProps<T>) => {
+    const [isOpen, setIsOpen] = useState(false);
+
+    const handleClose = () => {
+        if (isOpen) {
+            setIsOpen(false);
+        }
+    };
+    const handleSelectRadio = (id: string) => {
+        const updatedOption = options.find((option) => option.id === id);
+        setTimeout(handleClose, 100);
```

Comment:
I would prefer a css transition to make this smoother if that is the intention

Reply:
Elastic doesn't hide the Popover but removes it from the DOM, so I think It won't work with css transition.